pkg/controller/controlplane/valuesprovider.go

```diff
@@ -759,37 +728,30 @@ func (vp *valuesProvider) getControlPlaneShootChartValues(
 	map[string]interface{},
 	error,
 ) {
-	k8sVersionLessThan119, err := version.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, "<", openstack.CSIMigrationKubernetesVersion)
-	if err != nil {
-		return nil, err
-	}
-
 	var (
 		cloudProviderDiskConfig []byte
 		userAgentHeader         []string
 		csiNodeDriverValues     map[string]interface{}
 		caBundle                string
 	)
 
-	if !k8sVersionLessThan119 {
-		secret := &corev1.Secret{}
-		if err := vp.Client().Get(ctx, kutil.Key(cp.Namespace, openstack.CloudProviderCSIDiskConfigName), secret); err != nil {
-			return nil, err
-		}
+	secret := &corev1.Secret{}
+	if err := vp.Client().Get(ctx, kutil.Key(cp.Namespace, openstack.CloudProviderCSIDiskConfigName), secret); err != nil {
+		return nil, err
+	}
 
-		cloudProviderDiskConfig = secret.Data[openstack.CloudProviderConfigDataKey]
-		checksums[openstack.CloudProviderCSIDiskConfigName] = gardenerutils.ComputeChecksum(secret.Data)
-		userAgentHeader = vp.getUserAgentHeaders(ctx, cp, cluster)
+	cloudProviderDiskConfig = secret.Data[openstack.CloudProviderConfigDataKey]
+	checksums[openstack.CloudProviderCSIDiskConfigName] = gardenerutils.ComputeChecksum(secret.Data)
+	userAgentHeader = vp.getUserAgentHeaders(ctx, cp, cluster)
 
-		caSecret, found := secretsReader.Get(caNameControlPlane)
-		if !found {
-			return nil, fmt.Errorf("secret %q not found", caNameControlPlane)
-		}
-		caBundle = string(caSecret.Data[secretutils.DataKeyCertificateBundle])
+	caSecret, found := secretsReader.Get(caNameControlPlane)
+	if !found {
+		return nil, fmt.Errorf("secret %q not found", caNameControlPlane)
 	}
+	caBundle = string(caSecret.Data[secretutils.DataKeyCertificateBundle])
 
 	csiNodeDriverValues = map[string]interface{}{
-		"enabled":           !k8sVersionLessThan119,
+		"enabled":           true,
```

Comment:
We could enable the csi-driver-controller chart by default:
https://github.com/gardener/gardener-extension-provider-openstack/blob/1b146ec178cf4f273f41723fd47fa7d36c72bd51/charts/internal/seed-controlplane/values.yaml#L6-L7